simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,269 @@
+                    self.placesAffectingLocalThings[place][thing], self.localLikesVariance, 0, 1)
+                likeability[thing] += truncnorm_gen.rvs()
+            likeability[thing] /= person.lived.size
+        return likeability
+
+    def __flipCoin(self, bias=0.5):
+        """Return 1 or 0 based on a biased coin toss. Default bias = 0.5."""
+        return 1 if np.random.random() < bias else 0
+
+    def __calculateSimilarity(self, p1, p2):
+        """Return similarity measure between two persons (cosine similarity [0-1])."""
+        person1 = p1.lived
+        person1 = np.concatenate(
+            (person1, np.array(list(p1.globalLikes.values()))), axis=None)
+        person1 = np.concatenate(
+            (person1, np.array(list(p1.localLikes.values()))), axis=None)
+        person1 = person1.reshape((1, -1))
+
+        person2 = p2.lived
+        person2 = np.concatenate(
+            (person2, np.array(list(p2.globalLikes.values()))), axis=None)
+        person2 = np.concatenate(
+            (person2, np.array(list(p2.localLikes.values()))), axis=None)
+        person2 = person2.reshape((1, -1))
+
+        return cosine_similarity(person1, person2)
+
+    def __generatePeople(self):
+        """Generate list of people with their feature matrices filled in."""
+        people = []
+        for i in range(self.noOfPeople):
+            person = Person()
+
+            person.index = i
+            truncnormGenerator = get_truncated_normal(
+                self.placesLivedMean, self.placesLivedSD, 1, self.placesLivedUpper)
+            noOfPlacesLived = int(truncnormGenerator.rvs())
+
+            person.lived = np.zeros(self.places.size).reshape(1, -1)
+            person.lived[0][np.random.choice(
+                self.places, noOfPlacesLived, replace=False).tolist()] = 1
+
+            person.globalLikes = dict(
+                zip(self.globalThings, np.random.uniform(0, 1, self.globalThings.size)))
+            person.localLikes = self.__getLikability(person)
+            people.append(person)
+        return people
+
+    def __generateKnows(self, people):
+        """Return the knows truth matrix for all people in the dataset."""
+        knowsMatrix = np.zeros((self.noOfPeople, self.noOfPeople))
+        for pair in combinations(people, 2):
+            similarity = self.__calculateSimilarity(pair[0], pair[1])
+            knows = self.__flipCoin(similarity)
+
+            # set both (i,j) and (j,i) since P1 knows P2 => P2 knows P1.
+            knowsMatrix[pair[0].index][pair[1].index] = knows
+            knowsMatrix[pair[1].index][pair[0].index] = knows
+
+        return knowsMatrix
+
+    def __writeDataToFile(self, data, paths):
+        """Write data to file."""
+        # We have the same keys for data and its corresponding path.
+        for key in data:
+            with open(paths[key], 'w') as f:
+                for dataItem in data[key]:
+                    f.write(dataItem)
```

Comment:
The data into this method should be something like a list of lists.
"Data should be stored as data."
```
"\t".join([str(val) for val in a]) + "\n"
```